docs/realm.js

```diff
@@ -307,16 +307,23 @@ class Realm {
   compact() {}
 
   /**
-   * Writes a compacted copy of the Realm to the given path.
+   * Writes a compacted copy of the Realm α) to the given path or β) with the given output
+   *
+   * For invocation with α):
+   *   * Input Realms may be local or synced, encrypted or non-encrypted
+   *   * Output Realms will be local only, encrypted or non-encrypted
+   *
+   * For invocation with β):
+   *   * Input Realms may be local or synced, encrypted or non-encrypted
+   *   * Output Realms will be local or synced, encrypted or non-encrypted, depending on the configuration passed to the function
```

Comment:
I think we should have some examples of what one could achieve with this function.  

- encrypt unencrypted realm
- attempt to make a smaller realm (compaction)
- convert a local realm to a sync-realm

And as stated below, focus on the `configuration` as the input argument and perhaps make a note that `(path, encryptionKey)` is deprecated and support may be removed in the next major release.

Reply:
We should also document that the configuration must contain a `path` and will perform encryption if given an `encryptionKey`

Reply:
I think these examples are better left to the docs.  I assume a user would land on this method trying to implement conversions with guidance from documentation or customer support.